src/main/java/commands/DeadlineCommand.java

```diff
@@ -0,0 +1,44 @@
+package commands;
+
+import console.InputParser;
+import error.DukeTaskNameEmptyException;
+import error.Error;
+import task.TaskManager;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * Add a new 'deadline' type task to TaskManager.
+ */
+public class DeadlineCommand extends Command {
+
+    public static final String COMMAND_WORD = "deadline";
+
+    /**
+     * Creates a new deadline command and set the TaskManager object to perform operations on.
+     *
+     * @param taskManager The TaskManager object to execute commands on.
+     */
+    public DeadlineCommand(TaskManager taskManager) {
+        super(taskManager);
+    }
+
+    /**
+     * Add a new 'deadline' task.
+     *
+     * @return The deadline command type.
+     */
+    @Override
+    public String executeCommand() {
+        try {
+            taskManager.addDeadlineTask(InputParser.getTaskDetails(commandComponents));

```

Comment:
This line is a bit long. You can parse the input first and then add the deadline.